kayak_core/src/layout.rs

```diff
@@ -43,17 +44,20 @@ impl From<Rect> for Layout {
             height: rect.height,
             x: rect.posx,
             y: rect.posy,
-            z_index: rect.z_index,
+            z: rect.z_index,
         }
     }
 }
 
 ///
-/// Struct used for
+/// Struct used for [crate::OnLayout] as layout event data.
 ///
 pub struct LayoutEvent {
+    /// Layout of target component
     pub layout: Layout,
+    /// Flags noting one of dimensions has changed.
```

Comment:
We can probably be more generic here:
```suggestion
    /// Flags denoting the layout change.
```